### DISCUSSION TITLE
Flows redirections for dynamic environments

### DISCUSSION THREAD

**Discussion by alexandrebrg** [ory/kratos#1666] [Q&A] [answered]
Hello,

we are currently developing a client-side application with Kratos, which is working pretty well, but after updating the configuration I thought of a problem which might not be handled by Kratos yet to match out Gitflow:

We have a production environment and multiple feature environments (which are dynamically created by our GitFlow), we considered to have only one Kratos server for production and one kratos to handle **every** feature environments.  We are on GitLab, so one MR = one feature environment deployed.

The problem with this, is that every redirection URL use a single domain in the configuration examples. So my question is, if we only put a path like `/u/signup`, will it understand that it will redirect to the origin host with the redirection path ? If not, is there a another way to handle this case ? 

Thnaks!

**Comment by aeneasr** [comment]
Multi-domain environments are currently not well supported unfortunately.I think your solution could work but it's not officially supported.

**Reply by thomasboni** [reply]
Hello @AlexandreBrg @aeneasr , I am testing the workaround by replacing full URL with only path (and suppose that Kratos will use origin host URL as suggested). My configuration:

```
    selfservice:                                              
      default_browser_return_url: /
      flows:                                                  
        error:                                                
          ui_url: /                                           
        login:                                                
          lifespan: 10m                                       
          ui_url: /u/signin                                   
        registration:                                         
          after:                                              
            oidc:                                             
              hooks:                                          
              - hook: session                                 
          lifespan: 10m                                       
          ui_url: /u/signup                                   
        verification:                                         
          enabled: true                                       
          ui_url: /u/verify                                   
```

Unfortunately, this doesn't work. Kratos requires valid URL in each `ui_url` parameter:

```
level=fatal msg=Configuration value from key selfservice.flows.login.ui_url is not a valid URL: /u/signin audience=application reason=expected scheme to be set service_name=Ory Kratos service_version=v0.7.1-al
```

Do you have any idea to make this workaround work ?

**Comment by emrahcom** [comment, marked as answer]
I'm new to Kratos, my answer may be completely wrong but [this doc](https://www.ory.sh/kratos/docs/concepts/browser-redirect-flow-completion/) seems related to me. `return_to` and `whitelisted_return_urls` parts...

**Reply by alexandrebrg** [reply]
You might be right, it seems to be related to add `?return_to=https://www.myapp.com/blog/write` when requesting the flow.  Must be tested, but this solution requires an adaptation from the client-side application and not from Kratos.

I will try it, and will update this discussion accordingly

**Reply by alexandrebrg** [reply]
Sorry for the delay, but after some time I am able to confirm that this redirection works properly, only detail is that `return_to` query param has to be put into the initialize request & not in the submission.
